.github/workflows/autolabel-pullrequests.yml

```diff
@@ -65,58 +79,85 @@ jobs:
                 console.log("file changed under OPENWRT component : " + f['filename']);
-                lbls.add("component: openwrt");
+                if(!curLbls.has("component: openwrt")) {
+                  newLbls.add("component: openwrt");
+                }
                 break;
               case /^lte\/gateway\/.*/.test(f.filename):
               case /^lte\/protos\/.*/.test(f.filename):
               case /^orc8r\/gateway\/c\/.*/.test(f.filename):
               case /^third_party\/gtp_ovs\/.*/.test(f.filename):
                 console.log("file changed under AGW component : " + f['filename']);
-                lbls.add("component: agw");
+                if(!curLbls.has("component: agw")) {
+                  newLbls.add("component: agw");
+                }
                 break;
+              case /.*\/cloud\/.*/.test(f.filename):
               case /^\.golangci\.yml/.test(f.filename):
               case /^orc8r\/.*/.test(f.filename):
                 console.log("file changed under ORC8R component : " + f['filename']);
-                lbls.add("component: orc8r");
+                if(!curLbls.has("component: orc8r")) {
+                  newLbls.add("component: orc8r");
+                }
                 break;
               case /^xwf\/.*/.test(f.filename):
                 console.log("file changed under XWF component : " + f['filename']);
-                lbls.add("component: xwf");
+                if(!curLbls.has("component: xwf")) {
+                  newLbls.add("component: xwf");
+                }
                 break;
               case /^show-tech\/.*/.test(f.filename):
                 console.log("file changed under SHOW-TECH component : " + f['filename']);
-                lbls.add("component: show-tech");
+                if(!curLbls.has("component: show-tech")) {
+                  newLbls.add("component: show-tech");
+                }
                 break;
               case /^docs\/.*/.test(f.filename):
                 console.log("file changed under DOCS: " + f['filename']);
-                lbls.add("component: docs");
+                if(!curLbls.has("component: docs")) {
+                  newLbls.add("component: docs");
+                }
                 break;
              } // end of switch case
             } // end of for loop
 
-            if( lbls.size == 0 )  {
-              console.log("No components changed in this PR: " + context.issue.number);
+            if( newLbls.size == 0 )  {
+              console.log("No new component files changed in this PR: " + context.issue.number);
             }
             else  {
-              let uniqLbls = Array.from(lbls);
+              for (const l of curLblObjs['data']) {
+                if(!(l['name'].startsWith("component: ")))  {
+                  console.log("Adding Label:" + l['name']);
+                  newLbls.add(l['name']);
+                }
+              }
 
+              let uniqLbls = Array.from(newLbls);
```

Comment:
you are still using the set API, please only post the difference aka use put or delete API and only use set when the current labels are empty.